include/evmc/evmc.h

```diff
@@ -170,8 +170,8 @@ struct evmc_message
     /**
      * The optional value used in new contract address construction.
      *
-     * Needed only for a Host to calculate created address when kind is ::EVMC_CREATE2.
-     * Ignored in evmc_execute_fn().
+     * Needed only for a Host to calculate created address when kind is ::EVMC_CREATE2,
+     * ::EVMC_CREATE3 or ::EVMC_CREATE4. Ignored in evmc_execute_fn().
```

Comment:
From the point of view of EVM-Host interaction CREATE3 and CREATE4 executions are the same, so I thought to cover both with a singe kind `EVMC_CREATE3`

Reply:
OK, finally got to this. `EVMC_CREATE4` was introduced earlier, when `CREATE4` recursive validation was turned on.

At the moment we do hard failure (consume all gas) on eof validation failure (tentatively to be changed to light failure, but this is still pending discussion and decision). Hard failure is done more naturally in the host and requires `EVMC_CREATE4`. We need to finally decide on the light failure change and then I'll revert this addition.

But it can be done regardless of the lookup by hash change in the target `create4` branch.